javascript/saveFavoriteCity.js

```diff
@@ -0,0 +1,61 @@
+import { onPlaceChanged } from "./searchInput";
+import { displayFavoriteCities } from "./displayFavoriteCities";
+import { API } from "./api";
+
+export const saveFavoriteCity = (lat, lng, currentCityName) => {
+  const favoriteButton = document.querySelector(".favorite-btn");
+  const fav = document.getElementById("fav");
+
+  favoriteButton.addEventListener("click", async function () {
+    let favorites = JSON.parse(localStorage.getItem("favorites")) || [];
+
+    const selectedCityDataFromInput = onPlaceChanged();
+    let selectedCityData;
+
+    if (selectedCityDataFromInput) {
+      selectedCityData = selectedCityDataFromInput;
+    } else {
+      const dropdown = document.getElementById("favorite-city-dropdown");
+      const selectedCityName = dropdown.value;
+
+      selectedCityData = favorites.find(
+        (city) => city.name === selectedCityName
+      );
+
+      if (!selectedCityData) {
+        try {
+          const { cityName, data } = await API.getCurrentCityName(lat, lng);
```

Comment:
what is this for?
`cityName` and `data` seem not to be used

Reply:
Oh yeah it's not used so it should be removed.

Reply:
Ok then I'll remove this code